### ISSUE TITLE
Sub-pixel level Algorithms

### ISSUE THREAD

**Issue by DICeuser22** [dicengine/dice#211]
The pixel-level algorithms used at DICe are "Zero-normalized sum squared differences (ZNSSD)" correlation criteria, but What are the sub-pixel level algorithms used?

**Comment by dicengine** [comment]
I'm not sure I understand your question. The ZNSSD criteria is simply a way (or norm) to measure the difference between the intensity values of two collections of pixels. I'm wondering what connection you are trying to make with sub-pixel accuracy, which has to do with many things in addition to the correlation criteria. One can certainly achieve sub-pixel accuracy with an algorithm that includes a ZNSSD correlation criteria.

**Comment by DICeuser22** [comment]
Yes, the ZNSSD criteria is simply a way (or norm) to measure the difference between the intensity values of two collections of pixels, but, if compression, traction, shear etc occurs in the pixels during the deformation, what algorithms are used to take that into account?

**Comment by dicengine** [comment]
Those are represented in the so called "shape functions" which are used to approximate the deformation. You might find the answers to your questions  in this report: https://github.com/dicengine/dice/blob/master/doc/reports/LocalGradientAlgorithm.pdf. In particular, the first page might address your curiosity about representing deformations in DIC.
